docs/guide/api-plugin.md

```diff
@@ -374,6 +374,9 @@ Vite plugins can also provide hooks that serve Vite-specific purposes. These hoo
       server?: ViteDevServer
```

Comment:
Tests can go into `playground/html` with a new plugin that uses this new property perhaps. e.g setting headers. We have a couple tests that tests `x-served-by` header for example.

`transformRequest` should not be affected. Those are use for JS mostly.

Yeah this would go into Vite 5. We don't usually backport features to 4.0